src/Spaces/dss_cuda.jl

```diff
@@ -427,13 +427,13 @@ function fill_send_buffer!(::ClimaComms.CUDADevice, dss_buffer::DSSBuffer)
     if nsend > 0
         nitems = nsend * nlevels * nfid
         nthreads, nblocks = _configure_threadblock(nitems)
-        CUDA.synchronize() # CUDA MPI uses a separate stream. This will synchronize across streams
+        CUDA.synchronize(;blocking=true) # CUDA MPI uses a separate stream. This will synchronize across streams
```

Comment:
@sriharshakandala is this line even needed? the previous kernel is on the same stream

Reply:
Originally, we just added the `synchronize` in 436. If I recall correctly, the addition `synchronize` was added in 430 has to do with `CUDA.jl` using a separate thread for MPI communications and making sure any earlier communication is completed!